metrotranscom/doorway
# feat: before app submission refresh access token

This PR addresses https://github.com/metrotranscom/doorway/issues/987


## Description
There are some edge cases where a public user could have their session voided at the time of app submission, but the UI isn't aware and allowed the submission to go through

This pr makes it so that we attempt a refresh of the access token so if the session is voided the user is given a heads up that their session expired and they are bounced

## How Can This Be Tested/Reviewed?
The most easily reproducible way to test this is:
- create a public user (userA)
- go through the app submission process but STOP just before you accept the terms and fully submit the record
- in an incognito window log in as that same user (userA)
- in the OG window hit submit
you should be prompted with a dialog and whether you hit the "x" or "OK" you will be bounced to the sign in page and logged out

you should also test that going through the regular app submission process still works

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
